File: fixtures/rust/get_user_agent.rs
extern crate reqwest;
use reqwest::header;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut headers = header::HeaderMap::new();
    headers.insert("x-msisdn", "XXXXXXXXXXXXX".parse().unwrap());
    headers.insert("User-Agent", "Mozilla Android6.1".parse().unwrap());

    let res = reqwest::Client::new()
        .get("http://205.147.98.6/vc/moviesmagic?p=5&pub=testmovie&tkn=817263812")
        .headers(headers)
        .send()?
        .text()?;
    println!("{}", res);

    Ok(())
}

File: fixtures/matlab/get_user_agent.m
%% Web Access using Data Import and Export API
params = {
    'p' '5'
    'pub' 'testmovie'
    'tkn' '817263812'
};
baseURI = 'http://205.147.98.6/vc/moviesmagic';
uri = [baseURI '?' char(join(join(params, '='), '&'))];
options = weboptions(...
    'UserAgent', 'Mozilla Android6.1',...
    'HeaderFields', {'x-msisdn' 'XXXXXXXXXXXXX'}...
);
response = webread(uri, options);

% As there is a query, a full URI may be necessary instead.
fullURI = 'http://205.147.98.6/vc/moviesmagic?p=5&pub=testmovie&tkn=817263812';
response = webread(fullURI, options);

%% HTTP Interface
import matlab.net.*
import matlab.net.http.*

params = {
    'p' '5'
    'pub' 'testmovie'
    'tkn' '817263812'
};
header = [
    HeaderField('x-msisdn', 'XXXXXXXXXXXXX')
    HeaderField('User-Agent', 'Mozilla Android6.1')
]';
uri = URI('http://205.147.98.6/vc/moviesmagic', QueryParameter(params'));
response = RequestMessage('get', header).send(uri.EncodedURI);
